### PR TITLE
Prepare docs for 10 release

### PR DESCRIPTION
**Pull request by marvinhagemeister**

### CODE REVIEW

File: content/en/guide/v10/components.md [hunk 1]
```diff
@@ -147,3 +147,33 @@ const Foo = <Fragment>foo</Fragment>;
...
 const Bar = <>foo</>;
 ```
+
+You can also return arrays from your components:
+
+```js
+function Columns() {
+  return [
+    <td>Hello</td>
+    <td>World</td>
```
Comment (jviide) [inline content/en/guide/v10/components.md]: Should there be a comma separating the tags?

File: content/en/guide/v10/whats-new.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+}
+```
+
+## CSS Custom Properties
+
+Sometimes it's the little things that make a huge difference. With the recent advancements in CSS you can leverage variables for styling:
+
+```jsx
+function Foo(props) {
+  return <div> style={{ --theme-color: "blue" }}>{props.children}</div>;
```
Comment (developit) [inline content/en/guide/v10/whats-new.md]:
```suggestion
  return <div> style={{ '--theme-color': 'blue' }}>{props.children}</div>;
```

File: content/en/guide/v10/whats-new.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        <ThemedButton />
+      </SomeComponent>
+    <Theme.Provider />
+  );
+}
+```
+
+## CSS Custom Properties
+
+Sometimes it's the little things that make a huge difference. With the recent advancements in CSS you can leverage variables for styling:
```
Comment (developit) [inline content/en/guide/v10/whats-new.md]:
```suggestion
Sometimes it's the little things that make a huge difference. With the recent advancements in CSS you can leverage [CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) for styling:
```

File: content/en/guide/v10/whats-new.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+```jsx
+function Foo(props) {
+  return <div> style={{ --theme-color: "blue" }}>{props.children}</div>;
+}
+```
+
+## Compat lives in core
+
+Although we were always keen on adding new features and pushing Preact forward, the `preact-compat` package didn't receive as much love. Up until now it has lived in a separate repository making it harder to introduce breaking changes.
```
Comment (developit) [inline content/en/guide/v10/whats-new.md]:
```suggestion
Although we were always keen on adding new features and pushing Preact forward, the `preact-compat` package didn't receive as much love. Up until now it has lived in a separate repository making it harder to coordinate large changes spanning Preact and the compatibility layer. By moving compat into the same package as Preact itself, there's nothing extra to install in order to use libraries from the React ecosystem.
```

File: content/en/guide/v10/whats-new.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+[Compat](/guide/v10/differences-to-react#features-exclusive-to-preactcompat) has learned several new tricks such as `forwardRef`, `memo` and countless compatibility improvements.
```
Comment (developit) [inline content/en/guide/v10/whats-new.md]:
```suggestion
The compatibility layer is now called [preact/compat](/guide/v10/differences-to-react#features-exclusive-to-preactcompat), and has learned several new tricks such as `forwardRef`, `memo` and countless compatibility improvements.
```

File: content/en/guide/v10/whats-new.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
+---
+name: What's new in Preact X
+description: 'New features and changes in Preact X'
+---
+
+# What's new in Preact X
+
+Preact X is a huge step forward from Preact 8.x. We've rethought every bit and byte of our code and added a plethora of major features in the process. Same goes to for compatibility enhancements to support more third-party libraries.
```
Comment (jviide) [inline content/en/guide/v10/whats-new.md]: "Same goes to for" -> "Same goes for"?

File: content/en/guide/v10/whats-new.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+      Counter: {value}
+      <button onClick={increment}>Increment</button>
+    </div>
+  );
+}
+```
+
+## createContext
+
+The `createContext`-API is a true successor for `getChildContext()`. Whereas `getChildContext` is fine when you're absolutely sure to never change a value, it falls apart as soon as a component in-between the provider and consumer blocks an update via `shouldComponentUpdate` when it returns `false`. With the new context API this problem is now a thing of the past. It is a true pub/sub solution to deliver updates deep down the tree.
```
[the remark this answers is not in the document]
Comment (marvinhagemeister) [reply content/en/guide/v10/whats-new.md]: It's better than the current one, so I'll commit it :)

**Pull request merged**